src/js/actions/deviceActions.js

```diff
@@ -590,6 +592,28 @@ export const getSessionDetails = (sessionId, deviceId, userId, startDate, endDat
   );
 };
 
+export const deviceFileDownload = (deviceId, path) => () => GeneralApi.post(`${deviceConnect}/devices/${deviceId}/download`, { path });
+
+export const deviceFileUpload = (deviceId, path, file) => dispatch => {
+  var formData = new FormData();
+  formData.append('file', file);
+  formData.append('path', path);
+  const cancelSource = axios.CancelToken.source();
+  return Promise.all([
+    dispatch(setSnackbar('Uploading file')),
+    dispatch({ type: AppConstants.UPLOAD_PROGRESS, inprogress: true, uploadProgress: 0, cancelSource }),
+    GeneralApi.upload(`${deviceConnect}/devices/${deviceId}/upload`, formData, e => progress(e, dispatch), cancelSource.token)
```

Comment:
File upload is (now) implemented as PUT, not POST.
```suggestion
    GeneralApi.uploadPut(`${deviceConnect}/devices/${deviceId}/upload`, formData, e => progress(e, dispatch), cancelSource.token)
```

Reply:
This obviously requires something like:
```
diff --git a/src/js/api/general-api.js b/src/js/api/general-api.js
index 873d38ac3..b2fdfb39c 100644
--- a/src/js/api/general-api.js
+++ b/src/js/api/general-api.js
@@ -33,7 +33,8 @@ const Api = {
   post: (url, data) => authenticatedRequest.post(url, data),
   postUnauthorized: (url, data) => axios.post(url, data, commonRequestConfig),
   put: (url, data) => authenticatedRequest.put(url, data),
-  upload: (url, formData, progress, cancelToken) => authenticatedRequest.post(url, formData, { onUploadProgress: progress, timeout: 0, cancelToken })
+  upload: (url, formData, progress, cancelToken) => authenticatedRequest.post(url, formData, { onUploadProgress: progress, timeout: 0, cancelToken }),
+  uploadPut: (url, formData, progress, cancelToken) => authenticatedRequest.put(url, formData, { onUploadProgress: progress, timeout: 0, cancelToken })
 };

 export default Api;
 ```

Reply:
thanks for these => I took them as they were and it worked flawlessly 😃